File: apps/backend/qartnet/src/main/java/tn/enicarthage/qartnet/service/ChatbotService.java
package tn.enicarthage.qartnet.service;

import tn.enicarthage.qartnet.dto.request.AskChatbotRequest;
import tn.enicarthage.qartnet.dto.response.AskChatbotResponse;
import tn.enicarthage.qartnet.dto.response.ChatMessageResponse;

import java.util.List;

public interface ChatbotService {

    AskChatbotResponse ask(AskChatbotRequest request, String username);

    List<ChatMessageResponse> getHistory(String username);

    void resetConversation(String username);
}

File: apps/webapp/qartnet-frontend/src/app/features/ai-chatbot/services/chatbot.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable, inject } from '@angular/core';
import { Observable, map } from 'rxjs';
import { environment } from '../../../../environments/environment';
import { ApiResponse } from '../../../shared/models/api-response.model';
import {
  AskChatbotRequest,
  AskChatbotResponse,
  ChatMessageResponse,
} from '../models/chatbot.models';

@Injectable({ providedIn: 'root' })
export class ChatbotService {
  private http = inject(HttpClient);
  private api = `${environment.apiUrl}/api/ai/chat`;

  ask(message: string): Observable<AskChatbotResponse> {
    const req: AskChatbotRequest = { message };
    return this.http
      .post<ApiResponse<AskChatbotResponse>>(this.api, req)
      .pipe(map((res) => res.data!));
  }

  getHistory(): Observable<ChatMessageResponse[]> {
    return this.http
      .get<ApiResponse<ChatMessageResponse[]>>(`${this.api}/history`)
      .pipe(map((res) => res.data!));
  }

  reset(): Observable<void> {
    return this.http.post<void>(`${this.api}/reset`, {});
  }
}
